src/mixins/prevent-unsaved-mixin.ts

```diff
@@ -0,0 +1,55 @@
+import type { LitElement } from "lit";
+import type { Constructor } from "../types";
+import { isNavigationClick } from "../common/dom/is-navigation-click";
+import { navigate } from "../common/navigate";
+
+export const PreventUnsavedMixin = <T extends Constructor<LitElement>>(
+  superClass: T
+) =>
+  class extends superClass {
+    private _handleClick = async (e: MouseEvent) => {
+      const href = isNavigationClick(e);
+
+      if (!href) {
+        return;
+      }
+
+      e.preventDefault();
+      const result = await this.promptDiscardChanges();
+      if (result) {
+        navigate(href);
+      }
```

Comment:
Instead if copying this logic <https://github.com/home-assistant/frontend/blob/f97b48a51e051fcc3cff2dee0b1e07ac0bd0d4d6/src/layouts/home-assistant.ts#L135-L140> here, we could also re-fire the event and have the original code handle it

Reply:
I tried this as well, but couldn't get it work. I tried to re-fire the event after the confirm with `e.currentTarget.dispatchEvent(e);` or `window.dispatchEvent(e);`

Reply:
You will need to make a copy of the event:
```js
    const newEvent = new Event(oldEvent.type, oldEvent);
    e.currentTarget.dispatchEvent(newEvent);
```

Reply:
Unfortunately, it doesn't work either. I can see that the event is being dispatched and I can also catch it with a `mousedown` listener.

I used the `mousedown` listener here to avoid resetting the sidebar but I'm happy to change this

Reply:
Oh thats why it doesnt work, because the main uses a click event

Reply:
Nvm, I might have found a solution.